tests/selftest_browser_close.js

```diff
@@ -0,0 +1,18 @@
+const assert = require('assert').strict;
+const {newPage} = require('../src/browser_utils');
+const { onTeardown } = require('../src/runner');
+
+async function run(config) {
+    const page = await newPage(config);
+    await page.setContent('Hello world');
+
+    onTeardown(config, () => {
+        // console.log(page)
+        assert(page._closed, 'Page wasn\'t closed during teardown');
+    });
+}
+
+module.exports = {
+    description: 'Close browser automatically on teardown',
```

Comment:
This test sounds reasonable, but is rather unexpected in this PR. Was the file added by accident?

Reply:
Whoops, yup that was an accident. Looks like I'm switching between too many branches 🙈